File: Graphs/floyd_warshall.py
def floyd_warshall(graph, n):
    dist = [row.copy() for row in graph]
    for k in range(n):
        for i in range(n):
            for j in range(n):
                dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])
    return dist

def main():
    n, m = map(int, input().split())
    graph = [[float("inf") for _ in range(n)] for _ in range(n)]
    for _ in range(m):
        u, v, p = map(int, input().split())
        graph[u - 1][v - 1] = p
        graph[v - 1][u - 1] = p
    dist = floyd_warshall(graph, n)
    print(dist)

if __name__ == '__main__':
    main()

File: Competitive_Programming/Graphs/floyd_warshall.cpp
#include <bits/stdc++.h>
using namespace std;

#define MAXN 110

int n, graph[MAXN][MAXN], dist[MAXN][MAXN];

void floyd_warshall() {
    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            dist[i][j] = graph[i][j];
        }
    }

    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            for (int k = 1; k <= n; k++) {
                dist[j][k] = min(dist[j][k], dist[j][i] + dist[i][k]);
            }
        }
    }
}

int main() {

}